syntax = "proto3";

import "bob_emploi/frontend/api/job.proto";
import "bob_emploi/frontend/api/options.proto";

package bayes.bob;
option go_package="bayes.org/bob_emploi/frontend/api/training";

message Training {
  // Main name of the training.
  string name = 1;

  // Url where the training is available.
  string url = 2 [(string_format) = URL_FORMAT];

  // Name of the city in which the training site is.
  string city_name = 3;

  // FORMACODEs associated to this training (cf http://formacode.centre-inffo.fr/?).
  repeated string formacodes = 4;

  // Hiring potential after training: 0 means unknown, then the scale is from 1 to 5.
  int32 hiring_potential = 5;

  // IDs for the job groups this training is relevant for.
  repeated string rome_ids = 6;

  // The level of studies needed for this training, if any.
  DegreeLevel degree_level = 7;

  // A set of filter IDs: do not show the training if one of these filters return
  // false for the training.
  repeated string filters = 8 [(string_format) = SCORING_MODEL_ID];
}

message Trainings {
  repeated Training trainings = 1;

  TrainingAgency vocational_rehabilitation_agency = 2;
}

message TrainingAgency {
  // URL of the agency's website.
  string url = 1 [(string_format) = URL_FORMAT];

  // ID of the region (state in the US) for which the agency is responsible.
  string region_id = 2;
}
